src/api/db/data/20200326170855_remove_obsolete_notifications.rb

```diff
@@ -0,0 +1,30 @@
+class RemoveObsoleteNotifications < ActiveRecord::Migration[5.2]
+  EVENT_TYPES = [
+    'Event::ReviewWanted',
+    'Event::RequestCreate',
+    'Event::RequestStatechange',
+    'Event::CommentForRequest',
+    'Event::CommentForProject',
+    'Event::CommentForPackage'
+  ].freeze
+
+  AFFECTED_EVENT_TYPES = [
+    'Event::ReviewWanted',
+    'Event::RequestCreate',
+    'Event::RequestStatechange',
+    'Event::CommentForRequest'
+  ].freeze
+
+  def up
+    delete_obsole_notifications
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  def delete_obsole_notifications
+    Notification.where(notifiable_type: nil, notifiable_id: nil, event_type: EVENT_TYPES).delete_all
+    Notification.where(event_type: AFFECTED_EVENT_TYPES).delete_all
```

Comment:
Not sure I follow. EVENT_TYPES is old events and AFFECTED_EVENT_TYPES is events that have wrong data?

Reply:
` EVENT_TYPES` means all the events we can create notifications to. I could rename it to `NOTIFIABLE_EVENT_TYPES`.

`AFFECTED_EVENT_TYPES` includes those with wrong data so we have to tackle in this data migration.

The two queries correspond to the two first steps wee planned: 

1. _Remove all Notifications without notifiable_. So I assumed all of the notifiable ones with notifiable set to nil.
2. _Remove all Notifications with `CommentForRequest`, `RequestCreate`, `RequestStatechange`, `ReviewWanted`._ What means the notifiable ones except `CommentForProject` and `CommentForPackage` with notifiable set.

Reply:
the naming is rather confusing. I would suggest EVENT_TYPES_TO_CHECK_FOR_NOTFIABLE and EVENT_TYPES_TO_REMOVE or something

Reply:
It will be EVENT_TYPES_TO_CHECK_FOR_NOTIFIABLE and  EVENT_TYPES_TO_BE_REMOVED, if are ok.